core/unit_test/TestGraph.hpp

```diff
@@ -539,6 +539,17 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), empty_graph_default_host_exec) {
   graph.get_execution_space().fence();
 }
 
+// Ensure that the graph always stores the node.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), node_lifetime) {
+  ASSERT_EQ(count.use_count(), 1);
+  const auto graph =
+      Kokkos::Experimental::create_graph(ex, [&](const auto& root) {
+        const auto node =
+            root.then_parallel_for(1, count_functor{count, bugs, 0, 0});
+      });
+  ASSERT_EQ(count.use_count(), 2);
```

Comment:
Should we be attempting to setup a problem that actually executes the graph and would fail if the lifetime is not properly extended?
Are we just cutting a corner here?

Reply:
The only way the reference counting can be 2 after the closure used to fill the graph is closed is if the node is stored. IMO there is no need to run the graph.

Moreover, if we were running the graph, it is not even guaranteed that it would fail since the memory pointed to by `count` is not necessarily wiped out even after a call to free (`e.g.`  see description of #7365 for `Cuda`).

Reply:
Yeah good enough for me I see that this fails as expected without the fix. The only other thing we could do is to assign a default constructed view to a view captured in the graph, after the graph is created and check during that assignment via tools that no deallocation happens.

Reply:
I like the idea. I can rework the test a bit tomorrow if you want.

Reply:
@crtrott I reworked the test. The functor is a view that I'm building in-place. I then check that the view is deallocated when the graph is destroyed.